### PR TITLE
RTEMIS-134: Added the label for the Location field.

### PR DESCRIPTION
**Pull request by RohanSingh-inno**
Ticket # : https://innoraft.atlassian.net/browse/RTEMIS-134

---
### What the PR does
`(Explain fixes, changes and impacts)`
- Added the label for location field.

---
###Extras
image

### CODE REVIEW

File: modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php [hunk 1]
```diff
@@ -355,6 +357,66 @@ public function updateSiblingListElement(array &$form, FormStateInterface $form_
...
 
+  /**
+   * Callback to fill the `field_location` label values.
+   */
+  public function alterCshsLabels(array &$form, FormStateInterface $form_state) {
+    $miniNode = $form_state->getFormObject()->getEntity();
+    $currLocationId = $form_state->getValue('field_location')[0]['target_id'] ?? $miniNode->get('field_location')->getString() ?? NULL;
+    if ($currLocationId) {
+      $labels = [];
+      $term_storage = \Drupal::entityTypeManager()->getStorage('taxonomy_term');
```
Comment (jaswinsingh) [inline modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php:368]: Use DI here.

File: modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php [hunk 2]
```diff
@@ -355,6 +357,66 @@ public function updateSiblingListElement(array &$form, FormStateInterface $form_
...
+        $term = $term_storage->load($currLocationId);
+        if ($term) {
+          $adjacent_parent = $term_storage->loadParents($currLocationId);
+          $prev_adjacent_parent = $term_storage->loadParents(array_key_first($adjacent_parent));
+          $prev_adjacent_parent = reset($prev_adjacent_parent);
+          $type_of_area_value = $prev_adjacent_parent->get('field_type_of_area')->value ?? NULL;
+        }
+      }
+      // Get the core config.
+      $location_schema_config = \Drupal::config('rte_mis_core.settings')->get('location_schema');
```
Comment (jaswinsingh) [inline modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php:398]: use DI here

File: modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php [hunk 3]
```diff
@@ -355,6 +357,66 @@ public function updateSiblingListElement(array &$form, FormStateInterface $form_
...
+        }
+      }
+      // For the last field.
+      elseif (count($loadParent) == 5) {
+        $term = $term_storage->load($currLocationId);
+        if ($term) {
+          $adjacent_parent = $term_storage->loadParents($currLocationId);
+          $prev_adjacent_parent = $term_storage->loadParents(array_key_first($adjacent_parent));
+          $prev_adjacent_parent = reset($prev_adjacent_parent);
+          $type_of_area_value = $prev_adjacent_parent->get('field_type_of_area')->value ?? NULL;
```
Comment (jaswinsingh) [inline modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php:394]: Please add termInterface check

File: modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php (hunk 1 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php]: ok

File: modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php [hunk 4]
```diff
@@ -355,6 +357,66 @@ public function updateSiblingListElement(array &$form, FormStateInterface $form_
...
+        }
+      }
+      // For the next field just after categorization field.
+      elseif (count($loadParent) == 4) {
+        $term = $term_storage->load($currLocationId);
+        if ($term) {
+          $adjacent_parent = $term_storage->loadParents($currLocationId);
+          $adjacent_parent = reset($adjacent_parent);
+          $type_of_area_value = $adjacent_parent->get('field_type_of_area')->value ?? NULL;
+        }
```
[the remark this answers is not in the document]
Comment (RohanSingh-inno) [reply modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php]: ok

File: modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php (hunk 2 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php]: ok

File: modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php (hunk 3 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php]: ok

**Review by jaswinsingh — APPROVED** [review]

**Pull request merged**